AD A1 complete

Obiudaku opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

## What did you learn from the changes you have made?

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?

## How were these changes tested?

## A reference to a related issue in your repository (if applicable)

## Checklist
- [ ] I can confirm that my changes are working as intended

Commit:
AD A1 complete

tianyi21 reviewed (changes requested):
Hi Obinna Emmanuel! 

Thank you for completing and submitting your ADS A1:
* You implemented algorithms to identify the missing number in a provided list.
* You utlized the `set` data structure in your solution.
* You demonstrated how you approached the problem and analyzed your solution.

I noticed the following issue(s) with your submission. I would be appreciate if you could provide a short explanation for each of the following points and revise your submission to address these issue(s):
* In your example 2, when `input: lst = [0,1,2,3]` you stated that the output should be `-1`, i.e., no missing number. But you proposed solution give `[4]` instead. Could you please explain why you algorihm doesn't work as expected here?

Thanks,
Tianyi [LS]

Commit:
AD A1 reviewed

Obiudaku commented:
Adjusted n:

n is now determined by max(nums) to ensure the range of numbers in full_set matches the input's range.
This avoids including an unnecessary number beyond the maximum in the input.  

I believe this caused the 4 to appear, I have reviewed and fixed it in my revised commit. 

Thanks

tianyi21 reviewed (approved):
Hi Obinna Emmanuel!

Thank you for revising your submission. Just for your reference, `set` data structure doesn't have ordering, so techinically speaking, without sorting the list, you current approach doesn't guarantee that the results is in sorted order. But I can accept your current approach in this assignment. 

Also, please kindly remember to submit your assignment 2 when you're available.

Thanks,
Tianyi [LS]
